common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/ProductEntitlementMapping.kt

```diff
@@ -1,49 +1,50 @@
 package com.revenuecat.purchases.common.offlineentitlements
 
+import com.revenuecat.purchases.utils.optNullableString
 import org.json.JSONArray
 import org.json.JSONObject
 
 data class ProductEntitlementMapping(
-    val mappings: List<Mapping>
+    val mappings: Map<String, Mapping>
 ) {
     companion object {
-        private const val PRODUCTS_KEY = "products"
-        private const val PRODUCT_ID_KEY = "id"
+        private const val PRODUCT_ENTITLEMENT_MAPPING_KEY = "product_entitlement_mapping"
+        private const val PRODUCT_ID_KEY = "product_identifier"
+        private const val BASE_PLAN_ID_KEY = "base_plan_id"
         private const val ENTITLEMENTS_KEY = "entitlements"
 
         fun fromJson(json: JSONObject): ProductEntitlementMapping {
-            val productsArray = json.getJSONArray(PRODUCTS_KEY)
-            val mappings = mutableListOf<Mapping>()
-            for (productIndex in 0 until productsArray.length()) {
-                val productObject = productsArray.getJSONObject(productIndex)
+            val productsObject = json.getJSONObject(PRODUCT_ENTITLEMENT_MAPPING_KEY)
+            val mappings = mutableMapOf<String, Mapping>()
+            for (mappingIdentifier in productsObject.keys()) {
```

Comment:
I'm not sure what to call this, since it would be `productId:basePlanId` or `productId`. In `StoreProduct`, this would be the `id`...

Reply:
Alternatively I can suggest `productsObjectKey`, but `mappingIdentifier` works fine imo